### PR TITLE
feat: dtype parameter included for to_iceberg function

### PR DESCRIPTION
**Pull request by paulobrunheroto**
### Feature or Bugfix

- Feature

### Detail
- `dtype` parameter included for the `athena.to_iceberg` function to address possible issues related to mixed or undetermined data types

### Relates
- <URL or Ticket>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: b0a6a28457e608d361c1eb187a365aa2e2d540ad
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1c007b5a-b52d-4afa-8a45-52e762c6fa62%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: b0a6a28457e608d361c1eb187a365aa2e2d540ad
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=700489ae-fc1b-4dd9-8353-981a9171a9f3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by jaidisido — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 6f718d5db9ea6f29f93d205257e7626d1f181c38
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=efc133a7-ac8b-45e7-9a33-59c976722839%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: b0a6a28457e608d361c1eb187a365aa2e2d540ad
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=acd1d10d-a625-4bd1-bebf-08a2c93ff5cd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 6f718d5db9ea6f29f93d205257e7626d1f181c38
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1a53eac9-a406-42e5-a225-a330e834791a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: b0a6a28457e608d361c1eb187a365aa2e2d540ad
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b853291e-47f0-4a52-984f-4c1efa95c386%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]
Thank you!

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 52e7a00aa241c300b9c425a92de1a46970c9716c
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=85a69af2-6735-4ca8-bf26-9464404ee99f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 52e7a00aa241c300b9c425a92de1a46970c9716c
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f676ca5a-c864-4273-a427-346a7ce739bd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: awswrangler/athena/_write_iceberg.py [hunk 1]
```diff
@@ -86,6 +86,7 @@ def to_iceberg(
     boto3_session: Optional[boto3.Session] = None,
     s3_additional_kwargs: Optional[Dict[str, Any]] = None,
     additional_table_properties: Optional[Dict[str, Any]] = None,
+    dtype: Optional[Dict[str, str]] = None,
```
Comment (jaidisido) [inline awswrangler/athena/_write_iceberg.py:89]: The `dtype` argument must also be handled in the `_create_iceberg_table` function below (i.e. when creating an Iceberg table).

Could you also please add a unit test to check that we have not missed anything else besides the above?

**Review by jaidisido — CHANGES_REQUESTED** [review]
Realized that we might be missing edge cases here and some unit testing would help. Thanks

File: awswrangler/athena/_write_iceberg.py (hunk 1 above)
Comment (paulobrunheroto) [reply awswrangler/athena/_write_iceberg.py:90]: That's right, I have forgot about the scenario of new tables... I believe that it should be adjusted now...

I have also tried to add an unit test similar to the `s3.to_parquet`, I hope it works...

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 3ce46c7d4fdde80caaf1296ee72fcfb8aab0a9cc
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e1dd32d3-7039-4ff5-8c40-97b8870b8c69%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 3ce46c7d4fdde80caaf1296ee72fcfb8aab0a9cc
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=35b692db-a870-47bc-a5de-fe7eb2565bf8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: b468afee76b39e4b823702a56673e37bdb117c58
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a32643c3-4e8a-450c-97c2-ebf4befc4dda%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: b468afee76b39e4b823702a56673e37bdb117c58
* Result: FAILED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4a101071-8c78-4453-bb91-99ba54824001%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 2d88f0c537d4aaeac6091f24848bd7804b2ef98a
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=44c0cbb2-10ff-44eb-b4e1-e732930c2f6d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubDistributedCodeBuild6-jWcl5DLmvupS
* Commit ID: 2d88f0c537d4aaeac6091f24848bd7804b2ef98a
* Result: SUCCEEDED
* [Build Logs](https://p0rshq9q5i.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c2ec429a-4788-4652-b992-9127ec18c869%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by jaidisido — APPROVED** [review]
I have fixed the test, looking good now, thank you!

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: 50c24d00b8645dab5259f839bf6f0d77bb8122a5
* Result: FAILED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c1b7ad9c-25d0-4733-98ca-d38c6daa24df%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by paulobrunheroto** [comment]
> I have fixed the test, looking good now, thank you!

Thanks for the fixes!

**Review by kukushking — APPROVED** [review]

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: GitHubCodeBuild8756EF16-4rfo0GHQ0u9a
* Commit ID: f83a4e73b061022db0742e4f88d0999385f1b728
* Result: SUCCEEDED
* [Build Logs](https://gqgcj9s2gc.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e3ce6032-5457-436b-a870-5877c35e1dfb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*
